src/netlink.c

```diff
@@ -0,0 +1,387 @@
+
+/* netlink object methods */
+static luaL_Reg netlink_methods[] = {
+    {"__gc",        meth_close},
+    {"__tostring",  auxiliar_tostring},
+    {"bind",        meth_bind},
+    {"close",       meth_close},
+    {"getfd",       meth_getfd},
+    {"send",        meth_send},
+    {"sendto",      meth_sendto},
+    {"receivefrom", meth_receivefrom},
+    {"receive",     meth_receive},
+    {"setfd",       meth_setfd},
+    {"settimeout",  meth_settimeout},
+    {"gettimeout",  meth_gettimeout},
+    {"setpeername", meth_setpeername},
+    {"getpeername", meth_getpeername},
+    {"getsockpid",  meth_getsockpid},
+    {NULL,          NULL}
+};
+
+/* functions in library namespace */
+static luaL_Reg func[] = {
+    {"netlink", global_create},
+    {NULL, NULL}
+};
+
+/*-------------------------------------------------------------------------*\
+* Initializes module
+\*-------------------------------------------------------------------------*/
+int netlink_open(lua_State *L) {
+    	/* create classes */
+    	auxiliar_newclass(L, "netlink{connected}", netlink_methods);
+    	auxiliar_newclass(L, "netlink{unconnected}", netlink_methods);
+    	/* create class groups */
+    	auxiliar_add2group(L, "netlink{connected}", "netlink{any}");
+    	auxiliar_add2group(L, "netlink{unconnected}", "netlink{any}");
+	
+	auxiliar_add2group(L, "netlink{connected}", "select{able}");  
+	auxiliar_add2group(L, "netlink{unconnected}", "select{able}");  
+	luaL_setfuncs(L, func, 0);
+	return 0;
+}
+
+/*-------------------------------------------------------------------------*\
+* Send data through connected netlink socket
+\*-------------------------------------------------------------------------*/
+static int meth_send(lua_State *L) {
+	p_netlink nl = (p_netlink) auxiliar_checkclass(L, "netlink{connected}", 1);
+	size_t payload_size, sent = 0;
+	const char *payload = luaL_checklstring(L, 2, &payload_size);
+	p_timeout tm = &nl->tm; 
+
+	int flags = luaL_optinteger(L, 3, 0);
+
+	struct nlmsghdr *nlh;
+
+	nlh = (struct nlmsghdr *)malloc(NLMSG_SPACE(payload_size));
+	if(nlh==NULL) {
+		lua_pushnil(L);
+		lua_pushliteral(L,"error allocating memory in send()");
+		return 2;
+	}
+	memset(nlh, 0, NLMSG_SPACE(payload_size));
+	nlh->nlmsg_len = NLMSG_SPACE(payload_size);
+	nlh->nlmsg_pid = nl->srcpid;
+	nlh->nlmsg_flags = flags;
+
+	memcpy(NLMSG_DATA(nlh), payload, payload_size);
+
+	int err = socket_send(&nl->fd, (char *)nlh, nlh->nlmsg_len, &sent, tm);
+	if(err!=IO_DONE){
+		lua_pushnil(L);
+		lua_pushliteral(L, "error sending message");
+		return 2;
+	}
+
+	lua_pushinteger(L, sent);
+	return 1;
+}
+
+/*-------------------------------------------------------------------------*\
+* Send data through unconnected netlink socket
+\*-------------------------------------------------------------------------*/
+static int meth_sendto(lua_State *L) {
+	p_netlink nl= (p_netlink) auxiliar_checkclass(L, "netlink{unconnected}",1);
```

Comment:
space before `=`, after comma. Also, be consistent in the spaces after a type cast, the malloc one above didn't have one, this one does.